libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/ExponentialHistogramUtils.java

```diff
@@ -25,28 +25,46 @@ public class ExponentialHistogramUtils {
 
     /**
      * Estimates the sum of all values of a histogram just based on the populated buckets.
+     * Will never return NaN, but might return +/-Infinity if the histogram is too big.
      *
      * @param negativeBuckets the negative buckets of the histogram
      * @param positiveBuckets the positive buckets of the histogram
-     * @return the estimated sum of all values in the histogram, guaranteed to be zero if there are no buckets
+     * @return the estimated sum of all values in the histogram, guaranteed to be zero if there are no buckets.
      */
     public static double estimateSum(BucketIterator negativeBuckets, BucketIterator positiveBuckets) {
+        assert negativeBuckets.scale() == positiveBuckets.scale();
         double sum = 0.0;
-        while (negativeBuckets.hasNext()) {
-            double bucketMidPoint = ExponentialScaleUtils.getPointOfLeastRelativeError(
-                negativeBuckets.peekIndex(),
-                negativeBuckets.scale()
-            );
-            sum += -bucketMidPoint * negativeBuckets.peekCount();
-            negativeBuckets.advance();
-        }
-        while (positiveBuckets.hasNext()) {
+        while (negativeBuckets.hasNext() || positiveBuckets.hasNext()) {
```

Comment:
I found the two separate loops easier to follow. Is the only reason you're using a single loop to track the highest bucket so that you know which sign to use for the infinity? If so, couldn't you store the max negative and positive index?